app/assets/javascripts/the_bridge_ng/controllers/marketplace_ctrl.js.coffee

```diff
@@ -30,5 +31,24 @@
       return product.integrated
     else
       return true
+
+  $scope.changeVote = (flag, product_id) ->
```

Comment:
This method shouldn't need a flag or a product id.

Rather send the product instance that is provided in the repeat block.